From 4ec1792148a5d31a890dbe86bde40946404199e8 Mon Sep 17 00:00:00 2001
From: Julia Reynolds <juliacr@google.com>
Date: Fri, 10 Jan 2025 07:43:44 -0800
Subject: [PATCH] Remove notification content from icon a11y

The content can be read when the full cards are visible

Test: StatusBarIconViewTest
Test: enable talkback, highlight icons
Flag: EXEMPT bug fix
Bug: 338024220
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:2465326a187f11fefde17e667e9cf3fd890cdb57)
Merged-In: Ic607ab91e77bc660e3fccbc2390dd60151798ad7
Change-Id: Ic607ab91e77bc660e3fccbc2390dd60151798ad7
---
 .../systemui/statusbar/StatusBarIconView.java        | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
index 91c08a062b54..5c0b2a873620 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/StatusBarIconView.java
@@ -601,17 +601,7 @@ public class StatusBarIconView extends AnimatedImageView implements StatusIconDi
             }
         }
 
-        CharSequence title = n.extras.getCharSequence(Notification.EXTRA_TITLE);
-        CharSequence text = n.extras.getCharSequence(Notification.EXTRA_TEXT);
-        CharSequence ticker = n.tickerText;
-
-        // Some apps just put the app name into the title
-        CharSequence titleOrText = TextUtils.equals(title, appName) ? text : title;
-
-        CharSequence desc = !TextUtils.isEmpty(titleOrText) ? titleOrText
-                : !TextUtils.isEmpty(ticker) ? ticker : "";
-
-        return c.getString(R.string.accessibility_desc_notification_icon, appName, desc);
+        return c.getString(R.string.accessibility_desc_notification_icon, appName, "");
     }
 
     /**
-- 
2.48.1.262.g85cc9f2d1e-goog

